minpy/nn/solver.py

```diff
@@ -178,14 +193,27 @@ def loss_func(*params):
 
         self.loss_history.append(loss.asnumpy())
 
+        if self.visualize:
+            grad_norm = 0
+
         # Perform a parameter update
         for p, w in self.model.params.items():
             dw = grads[p]
+            if self.visualize:
+                norm = dw ** 2
+                while not isinstance(norm, minpy.array.Number):
+                    norm = sum(norm)
```

Comment:
@jermainewang Only a separate GradientVisualizer object or class? No need for other things like loss or accuracy?